### ISSUE TITLE
Youtube downloads frequently fail

### ISSUE THREAD

**Issue by spasticus74** [ytdl-org/youtube-dl#31629]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
youtube-dl -x --audio-format mp3 --add-metadata -o '%(title)s|%(duration)s.%(ext)s' --verbose https://www.youtube.com/watch?v=y2O0dfFPIhM&pp=ygUMa29ybiBvaWxkYWxl
System config: []
[debug] User config: ['-x', '--audio-format', 'mp3', '--add-metadata', '-o', '%(title)s|%(duration)s.%(ext)s']
[debug] Custom config: []
[debug] Command-line args: ['--verbose', 'https://www.youtube.com/watch?v=y2O0dfFPIhM']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.9.2 (CPython) - Linux-5.15.84-v8+-aarch64-with-glibc2.31
[debug] exe versions: ffmpeg 4.3.5-0, ffprobe 4.3.5-0
[debug] Proxy map: {}
[youtube] y2O0dfFPIhM: Downloading webpage
ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1794, in _real_extract
    'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id') if owner_profile_url else None,
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

```

## Description

I've started to see an error I haven't encountered before "Unable to extract uploader id;". I've been getting it a lot in the last 2 or 3 days across a whole bunch of different videos on youtube

**Comment by dirkf** [comment]
> [x] I've searched the bugtracker for similar issues including closed ones

Reeeeeeeeeeeeeeeeally?

#30839

**Comment by libertyteeth** [comment]
You said "duplicate" and provided no link other than to the rules.  Can you update?  Thanks!

(Asking this because I arrived here via search today, happy to see the issue was entered; then later, saw that it was closed as a duplicate, but if you know it's a duplicate of N, you can let us know how to find N more easily.  Thanks dirkf!)

**Comment by dirkf** [comment]
Read the [linked guidance](https://github.com/ytdl-org/youtube-dl/issues/30839) again to find the existing **Known issue**, as OP should have done.

**Comment by libertyteeth** [comment]
You are the person who already found the duplicate bug, is that true?

If so: you can help many people out by linking the duplicate you already found.  Rather than asking us to duplicate your appreciated efforts.

To be clear, it's great to curate; even better, to be clearer.  Thank you, for curating, and for taking the time to respond.

I took the time to follow your above instructions so for others watching/reading: the duplicate issue is #31530

This only links it in the comments though.  I'm "mostly new here" but have experience with other bug databases, and know that some have the ability to provide links for the metadata of the bug, to indicate dependencies, duplicates, etc.  If that's possible here, and you can do that final step, it would be appreciated.  Thanks again.

P.S. I see two of the issues are about "shorts" videos.  If I may, I offer a regex that converts a "shorts" URL into a full-video URL, which I find a lot more useful (I can rewind, change volume): 

`/\/shorts\//\/watch?v=/`

I use that in a script I wrote to convert the URL and launch Firefox with it.  This conversation gave me the idea to add that parsing to my wrapper script for youtube-dl.  Thank you very, very much for this motivation!  It has improved my day and future, and hopefully others who are witnessing.

**Comment by dirkf** [comment]
Glad you found the link.

There is limited workflow support in the GH issue tracker for guiding people to the correct issue (eg, other trackers allow you to show a list of matching issues from the tracker when someone is opening an issue). So we ask people to read the pinned self-help issue, which now includes explicit advice on searching and a list of major issues that have been fixed in the master code but not in the latest release. If someone doesn't read the self-help, the issue template checklist still asks for a search of existing issues, which OP claimed to have done, but without finding the matching issue. Finally a bug in the site allows users to bypass the issue templates, the original motivation for the self-help issue.

If users bypass the self-help and create duplicate issues, the result is what I (and Vangelis66) describe [here](https://github.com/ytdl-org/youtube-dl/issues/31613#issuecomment-1437279563).

The shorts work-around was discovered and documented in the linked 'known issue', but thanks. If you've installed a yt-dl that fixes #31530, it will include the fix for the 'known issue', essentially extending the URL pattern for YouTube videos in a similar way to your regex above, so no wrapper is needed; and all the other 'known issues' too.
